operator/pkg/util/kubeconfig.go

```diff
@@ -38,3 +44,40 @@ func CreateBasic(serverURL, clusterName, userName string, caCert []byte) *client
 		CurrentContext: contextName,
 	}
 }
+
+func IsInCluster(hostCluster *operatorv1alpha1.HostCluster) bool {
```

Comment:
Can we add some comments about this? To explain how to decide whether it's an in-cluster installation.